### PR TITLE
Frontend Guide

### PR DESCRIPTION
**Pull request by TheLoneRonin**
### Description

- Goes over how to create GraphQL queries.

- Goes over how to create GraphQL mutations.

- Goes over how to integrate GraphQL into React functional components.

### CODE REVIEW

File: guides/FRONTEND.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+export const Component: FC = () => {
+  const [state, executeMutation] = useMutation(CreateMap);
+  const { fetching, error, data } = state;
+
+  async function insertMap(author, name) {
+    await executeMutation({ author, name });
+  }
+
+  return(
+    <button type="button" onClick={async () => await insertMap('Hello World', 'Test 1234')}>
```
Comment (pakokrew) [inline guides/FRONTEND.md:184]: Avoid putting asynchronous function in synchronous event handlers. Errors will be lost here. 

Better write something like 
```
function insertMap(author, name) {
   executeMutation({ author, name })
      .catch(console.error); // or stuff
  }
```

**Review by pakokrew — APPROVED** [review]
good one :)

File: guides/FRONTEND.md (hunk 1 above)
Comment (TheLoneRonin) [reply guides/FRONTEND.md:184]: That is true!
Technically, you can write a try/catch in the event handler.

But the example provided is wrong.
Thanks, will update.

Comment (pakokrew) [reply guides/FRONTEND.md]: React event handlers are synchronous. Putting a try/catch could be a solution but the function would still be asynchronous. If by any mischance it throws, it leads to an unhandled exception, which is not good

**Comment by TheLoneRonin** [comment]
@pakokrew can merge once you approve. (I do not have merge priveliges)

File: guides/FRONTEND.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+- Docker
+
+You will need to be able to start the backend with:
+
+```bash
+yarn docker:start
+```
+
+If you have difficulties running the backend, checkout the [Backend Guide](BACKEND.md).
```
Comment (hammadj) [inline guides/FRONTEND.md:17]: We dont need them to run the backend, they can just set the `NEXT_PUBLIC_GRAPHQL_URL` in their .env file to `https://hasura-7s6e.onrender.com/`

**Review by hammadj — CHANGES_REQUESTED** [review]
Should update to include instructions on running only the frontend without needing to spin up the backend. Basically just need to add a .env.local file in the web package: https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

**Comment by pakokrew** [comment]
> Should update to include instructions on running only the frontend without needing to spin up the backend. Basically just need to add a .env.local file in the web package: https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

Note to myself: The `.env.local` file must be in `packages/web` folder

**Review by pakokrew — APPROVED** [review]

**Pull request merged**